app/components/Search/SearchPage.js

```diff
@@ -25,6 +27,16 @@ class SearchPage extends Component<Props, State> {
     query: this.props.location.query.q || ''
   };
 
+  componentWillReceiveProps(nextProps: Props) {
+    // Make sure the selectedIndex is within 0 <= index < results.length:
+    const selectedIndex = Math.min(
```

Comment:
How about just:
```js
const selectedIndex = (selectedIndex < resullts.length) ? selectedIndex : 0;
// Or magic with lodash
const selectedIndex = has(result, selectedIndex) ? selectedIndex : 0;
```
The `min` and `max` makes it kinda hard to quickly understand the flow.

Or just reset the index to 0  when new search-results appear.